torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -6943,10 +6943,65 @@ def test_ddp_multiple_nested_unused_params_err_ignore_params(self):
             # certain parameters.
             self._test_ddp_multiple_nested_unused_params_error(ignore_sparse=True)
 
-        @unittest.skipIf(
-            BACKEND != "nccl" and BACKEND != "gloo",
-            "Only Nccl & Gloo backend support DistributedDataParallel",
-        )
+        @unittest.skipIf(BACKEND != 'nccl' and BACKEND != 'gloo',
+                         "Only Nccl & Gloo backend support DistributedDataParallel")
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_inference(self):
+            # tests that DDP module can be run on a single node with no_grad
+            # or eval setting and there is no hang.
+            rank = self.rank
+            torch.cuda.set_device(rank)
+            model = Net().cuda()
```

Comment:
Nit: I believe you can just create a for loop over two tuples of <model, input>, where the models are `Net().cuda()` and `nn.SyncBatchNorm`. This can save some duplicate code and improve the readability.